File: app-java/src/main/java/com/github/iojjj/app/java/StringView.java
package com.github.iojjj.app.java;

import android.content.Context;
import android.graphics.Color;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.util.AttributeSet;
import com.github.iojjj.app.core.BaseStringView;
import com.github.iojjj.bootstrap.adapters.adapter.AdapterUtils;
import com.github.iojjj.bootstrap.adapters.adapter.PagedAdapter;

public final class StringView extends BaseStringView {

    public StringView(@NonNull final Context context) {
        super(context, "Java");
    }

    public StringView(@NonNull final Context context, @Nullable final AttributeSet attrs) {
        super(context, attrs, "Java");
    }

    public StringView(@NonNull final Context context, @Nullable final AttributeSet attrs, final int defStyleAttr) {
        super(context, attrs, defStyleAttr, "Java");
    }

    @Override
    public void bind(@NonNull final PagedAdapter<?> adapter, final int position, final String item) {
        if (adapter.isFiltered()) {
            updateText(AdapterUtils.highlightQuery(item, adapter.getFilterQuery(), Color.RED));
        } else {
            updateText(item);
        }
        setActivated(adapter.<String>toTyped().getSelectionTracker().getSelection().contains(item));
    }
}

File: app-kotlin/src/main/kotlin/com/github/iojjj/app/kotlin/StringView.kt
package com.github.iojjj.app.kotlin

import android.content.Context
import android.graphics.Color
import android.util.AttributeSet
import com.github.iojjj.app.core.BaseStringView
import com.github.iojjj.bootstrap.adapters.adapter.PagedAdapter
import com.github.iojjj.bootstrap.adapters.adapter.highlightQuery

class StringView : BaseStringView {

    constructor(context: Context) : super(context, "Kotlin")
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs, "Kotlin")
    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr, "Kotlin")

    override fun bind(adapter: PagedAdapter<*>, position: Int, item: String) {
        if (adapter.isFiltered) {
            updateText(item.highlightQuery(adapter.getFilterQuery(), Color.RED))
        } else {
            updateText(item)
        }
        isActivated = adapter.toTyped<String>().selectionTracker.selection.contains(item)
    }
}